packages/vscode-extension/src/handlers.ts

```diff
@@ -244,6 +245,10 @@ export async function activate(): Promise<Result<Void, FxError>> {
 
         await refreshEnvTreeOnFileChanged(workspacePath, files);
       });
+
+      workspace.onDidSaveTextDocument(async (event) => {
+        await refreshEnvTreeOnFileContentChanged(workspacePath, event.uri.fsPath);
```

Comment:
Is this manually triggered when user save in vscode?

Reply:
It will be triggered automatically when the file is saved, no matter by Teams toolkit or User.